llvm/lib/Transforms/InstCombine/InstCombineSimplifyDemanded.cpp

```diff
@@ -1919,6 +1919,19 @@ Value *InstCombinerImpl::SimplifyDemandedVectorElts(Value *V,
 /// For floating-point classes that resolve to a single bit pattern, return that
 /// value.
 static Constant *getFPClassConstant(Type *Ty, FPClassTest Mask) {
+  // TODO: Support aggregate types that are allowed by FPMathOperator.
+  switch (Mask) {
+    case fcPosZero:
+    case fcNegZero:
```

Comment:
These cases aren't tested

Reply:
I've put cases that (I think) _should_ hit these, however, `computeKnownFPClass()` only handles `ConstantAggregateZero` (and that case is now fixed), so I'm not sure it's possible to get fcPosZero, fcNegZero, etc for a constant aggregate.